apps/src/aiTutor/views/AITutorFooter.tsx

```diff
@@ -38,73 +29,35 @@ const AITutorFooter: React.FC<AITutorFooterProps> = ({renderAITutor}) => {
   );
   const studentCode = sources[fileMetadata[activeTabKey]].text;
 
-  const hasCompilationError = useAppSelector(
-    state => state.javalabEditor.hasCompilationError
-  );
-  const hasRunOrTestedCode = useAppSelector(
-    state => state.javalab.hasRunOrTestedCode
-  );
-  const isRunning = useAppSelector(state => state.javalab.isRunning);
-  const validationPassed = useAppSelector(
-    state => state.javalab.validationPassed
-  );
-
   const dispatch = useAppDispatch();
 
-  const showCompilationOption =
-    !isRunning && hasRunOrTestedCode && hasCompilationError;
-  const showValidationOption =
-    hasRunOrTestedCode && !hasCompilationError && !validationPassed;
-
-  const handleSubmit = useCallback(
-    (actionType: AITutorTypesValue) => {
-      if (isWaitingForChatResponse) {
-        return;
-      }
-
-      let studentInput = '';
-      let event = '';
-
-      switch (actionType) {
-        case ActionType.COMPILATION:
-          studentInput = QuickActions[ActionType.COMPILATION];
-          event = EVENTS.AI_TUTOR_ASK_ABOUT_COMPILATION;
-          break;
-        case ActionType.VALIDATION:
-          studentInput = QuickActions[ActionType.VALIDATION];
-          event = EVENTS.AI_TUTOR_ASK_ABOUT_VALIDATION;
-          break;
-        case ActionType.GENERAL_CHAT:
-        default:
-          studentInput = userMessage;
-          event = EVENTS.AI_TUTOR_ASK_GENERAL_CHAT;
-          break;
-      }
+  const handleSubmit = useCallback(() => {
+    if (isWaitingForChatResponse) {
+      return;
+    }
 
-      const chatContext = {
-        studentInput,
-        studentCode,
-        actionType,
-      };
+    const chatContext = {
+      studentInput: userMessage,
+      studentCode,
+      actionType: ActionType.GENERAL_CHAT,
+    };
 
-      dispatch(askAITutor(chatContext));
-      setUserMessage('');
+    dispatch(askAITutor(chatContext));
+    setUserMessage('');
 
-      analyticsReporter.sendEvent(event, {
-        levelId: level?.id,
-        levelType: level?.type,
-      });
-    },
-    [userMessage, studentCode, isWaitingForChatResponse, level, dispatch]
-  );
+    analyticsReporter.sendEvent(EVENTS.AI_TUTOR_ASK_GENERAL_CHAT, {
+      levelId: level?.id,
+      levelType: level?.type,
+    });
+  }, [userMessage, studentCode, isWaitingForChatResponse, level, dispatch]);
 
   const disabled = !renderAITutor || isWaitingForChatResponse;
 
   const userMessageIsEmpty = userMessage.trim() === '';
```

Comment:
Again, non-blocking, but would it be better to trim when the value is set?

That'd be on line 71:
```
onChange={e => setUserMessage(e.target.value.trim())}
```

Then we wouldn't need to worry about calling trim again later and can just directly check against an empty string or `.length`.

Reply:
I see where you're coming from, but I don't want to prematurely trim the whitespace within the `onChange` of the text box because then the user can't put any spaces in their text. I endupwithsomethinglikethis 😬